src/algo_tx.c

```diff
@@ -160,6 +160,38 @@ map_kv_bin(uint8_t **p, uint8_t *e, char *key, uint8_t *valbuf, int vallen)
   return 1;
 }
 
+static int
+map_kv_params(uint8_t **p, uint8_t *e, char *key, struct asset_params *params)
+{
+  // Save original buffer in case we end up with a zero value
+  uint8_t *psave = *p;
+
+  encode_str(p, e, key, SIZE_MAX);
+
+  uint8_t *mapbase = *p;
+  *((*p)++) = FIXMAP_0;
+
+  mapbase[0] += map_kv_bin   (p, e, "am", params->metadata_hash, sizeof(params->metadata_hash));
+  mapbase[0] += map_kv_str   (p, e, "an", params->assetname, sizeof(params->assetname));
+  mapbase[0] += map_kv_str   (p, e, "au", params->url, sizeof(params->url));
+  mapbase[0] += map_kv_bin   (p, e, "c",  params->clawback, sizeof(params->clawback));
+  mapbase[0] += map_kv_uint64(p, e, "dc", params->decimals);
+  mapbase[0] += map_kv_bool  (p, e, "df", params->default_frozen);
+  mapbase[0] += map_kv_bin   (p, e, "f",  params->freeze, sizeof(params->freeze));
+  mapbase[0] += map_kv_bin   (p, e, "m",  params->manager, sizeof(params->manager));
+  mapbase[0] += map_kv_bin   (p, e, "r",  params->reserve, sizeof(params->reserve));
+  mapbase[0] += map_kv_uint64(p, e, "t",  params->total);
+  mapbase[0] += map_kv_str   (p, e, "un", params->unitname, sizeof(params->unitname));
```

Comment:
this is maybe not worth supporting, but if any of these string fields contains a null byte, you're out of luck for signing with the ledger since it will truncate the encoding of the string there (though decoding works as expected).

Reply:
Yeah, I think I'm OK with not supporting such strings.